client/webserver/websocket.go

```diff
@@ -77,16 +77,21 @@ func (s *WebServer) websocketHandler(conn ws.Connection, ip string) {
 		return s.handleMessage(cl, msg)
 	})
 
+	// Lock the clients map before starting the connection listening so that
+	// synchronized map accesses are guaranteed to reflect this connection.
+	// Also, ensuring only live connections are in the clients map notify from
+	// sending before it is connected.
+	s.mtx.Lock()
```

Comment:
It may seem that this is only for the assumption in `TestClientMap`, but it is a good test since a wsConn handler may be written to expect the client in the map and the handlers may be called as soon as the connection is up.